modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+                    		-\int_{\Gamma_R}\left(\kappa\nabla\lambda\cdot\boldsymbol{n}\right)\delta T\text{d}\Gamma
+                           	+\int_{\Gamma_D}\lambda\left(\kappa\nabla\delta T\cdot\boldsymbol{n}\right)~\text{d}\Gamma\\
+=&\int_{\Gamma_R}\left(\lambda h -\left(\kappa\nabla\lambda\cdot\boldsymbol{n}\right)\right)\delta T\text{d}\Gamma
+    				+\int_{\Gamma_D}\lambda\left(\kappa\nabla\delta T\cdot\boldsymbol{n}\right)~\text{d}\Gamma,
+\end{aligned}
+\end{equation}
+where $\text{d}G(T)/\text{d}T=h$ from [eqn:optimization_bcs] was used.
+Combining [eqn:delta_bodyLagrangian] and [eqn:delta_bcLagrangian] to get $\delta\mathcal{L}$ results in
+\begin{equation}\label{eqn:delta_Lagrangian}
+\begin{aligned}
+\delta\mathcal{L}=&\delta\mathcal{A}+\delta\mathcal{B}\\
+=&\int \left(\left(\nabla\cdot \kappa \nabla \lambda \right)
+	+  \sum_{i=1}^{N} \delta(x - x_i) \left( T - \bar{T} \right) \right)\delta T ~\text{d}\Omega\\
+    &+\int_{\Gamma_R}\left(\lambda h -\left(\kappa\nabla\lambda\cdot\boldsymbol{n}\right)\right)\delta T\text{d}\Gamma
+    	+\int_{\Gamma_D}\lambda\left(\kappa\nabla\delta T\cdot\boldsymbol{n}\right)~\text{d}\Gamma.
+\end{aligned}
+\end{equation}
+
+Stationarity of $\mathcal{L}$ would require $\delta\mathcal{L} = 0$ for all admissible $\delta T$. Setting each of the integrals in [eqn:delta_Lagrangian] results in the adjoint problem and its boundary conditions
+\begin{equation} \label{eqn:adjoint_problem}
+\boxed{
+\begin{aligned}
+\nabla\cdot \kappa \nabla \lambda +  \sum_{i=1}^{N} \delta (x - x_i)(T - \bar{T}) &=0, ~\text{in}~\Omega,\\
+\lambda  &= 0, ~\text{on}~\Gamma_D, \\
+\kappa \nabla \lambda \cdot \boldsymbol{n} & = \lambda h, ~\text{on}~{\Gamma_R}. \\
+\end{aligned}
+}
+\end{equation}
+Solving [eqn:adjoint_problem] comes down to adjusting the boundary conditions and load vector from the forward problem and re-solving.
+
+[comment2]: <> (% ----------------------------------------------------------------------------------------------------------%)
+
+## PDE Derivatives for Inversion id=sec:PDEDerivs
+
+In this section we will present derivatives for steady state heat conduction [eqn:optimization_problem] with respect to the force or material parameters.  For all of these examples, measurement data is taken at specific locations where the objective function can be represented by [eqn:objective_integral].  We will present the discrete forms of our PDE and its derivative which most closely matches the implementation that will be used in MOOSE.
+
+The discrete form of the PDE constraint for steady state heat conduction in [eqn:optimization_problem], $\hat{g}$, is given as
+
+\begin{equation}\label{eq:discretePDE}
+\hat{\text g}=\textbf{K}\hat{\text T} - \hat{\text f}=0,
+\end{equation}
+where $\textbf{K}$ is the Jacobian matrix, $\hat{\text T}$ and $\hat{\text f}$ are the discretized  temperature and residual vectors. Element-wise definitions of the terms are
```

Comment:
I think `f` is typically called as source vector.

Reply:
done